search/includes/classes/class-health.php

```diff
@@ -12,6 +12,8 @@
 class Health {
 	const CONTENT_VALIDATION_BATCH_SIZE    = 500;
 	const CONTENT_VALIDATION_MAX_DIFF_SIZE = 1000;
+	const CONTENT_VALIDATION_LOCK_NAME = 'vip_search_content_validation_ongoing';
+	const CONTENT_VALIDATION_LOCK_TIMEOUT = 900; // 15 min
```

Comment:
I kinda think this should scale with the batch size. If it gets set super high and take 15 mins to run through we could potentially get duplicates.

What do you think?

Reply:
I am not sure... there seems to be no downside to have the number set to the upper limit in such scaling always. In the happy case scenario, it gets removed when done anyway. If we think 15 mins is too low we can do like 20 or even 30 I guess.

No downside meaning it would block another try for the remainder of that time if the validation would die unexpectedly. But that might be a good thing as well, or?

BTW this whole locking is modeled on the indexing lock including the naming. I don't think we need anything more advanced here.

Reply:
Transients can be easily deleted via wp cli, so it's not a huge concern.

I don't think the indexing locks are particularly effective either for what it's worth since if you set the batch sizes really high you can bypass the locking.

My proposal is to just get the average time per-post of a higher batch, add a little to it, and then set the lock based on that value multiplied by batch size. This should allow us to safely handle whatever sized batches are handed in without too much added complexity.

If it doesn't seem necessary, we can ship as is 🤷 .

Reply:
> just get the average time per-post of a higher batch
Do you have suggestions on doing that? 

My guess would be that it would vary somewhat based on how complex records each site has. So even if we would experimentally measure the time on one site it won't work well for another. Also, this code handles all types of indexables and I am not sure if _posts_ time would map to _users_ for instance. We could build some sort of dynamic mechanism that would keep updating the average per site and store it as an option or something. But as I mentioned before, I feel that is overkill.

Reply:
It doesn't need to be perfect or overly complicated. Just run batches on a few sites, take the ones that take the longest per-post, add a bit to account for some sites being slower, and then just use that magic number plus basic multiplication against the batch size to get the lock length. Can always bump the magic number if it's too low over-time.

Reply:
Did some testing. 
Slow Site:
**Batch Size, Average Time (seconds)**
500, 0.5
1000, 1.3
2000, 10,24

Fast Site:
**Batch Size, Average Time (seconds)**
500, 0.1
2000, 0.3
20000, 3

Sooo, the differences are quite huge. Maybe we could then do 1 second per 100 docs? Does that sound reasonable?

Reply:
I added the suggested ^^